numcodecs/tests/test_zarr3.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import annotations
+from typing import Iterator
+
+
+import numpy as np
+import pytest
+import sys
+
+from numcodecs.registry import get_codec
+
+try:
+    from zarr.codecs.registry import get_codec_class
+    from zarr.array import Array
+    from zarr.common import JSON
+    from zarr.codecs import BytesCodec
+    from zarr.abc.store import Store
+    from zarr.store import MemoryStore, StorePath
+
+except ImportError:
+    pass
+
+
+pytestmark = pytest.mark.skipif(
+    sys.version_info < (3, 10), reason="zarr-python 3 requires Python 3.10 or higher"
+)
+
+
+@pytest.fixture
+def store() -> Iterator[Store]:
+    yield StorePath(MemoryStore(mode="w"))
+
+
+@pytest.mark.parametrize(
+    "codec_id", ["blosc", "lz4", "zstd", "zlib", "gzip", "bz2", "lzma", "shuffle"]
```

Comment:
These codecs are specifically the ones that zarr v3 needs support for? If so it might be more obvious if there's a `ZARR3_CODECS` list defined near the top of this file with this list.

Reply:
Not really. These are all the codecs that have a generic-enough config that we can use one parameterized test. Other codecs need specialized tests.